todo-app/src/App.js

```diff
@@ -1,13 +1,152 @@
-import React, { Component } from 'react';
+import React, { Component } from 'react'
+import todos from './todosTable'
+import Todo from './todo'
+import './App.css'
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      todos: [],
+      newTodo: {
+        id: '',
+        description: '',
+        stage: ''
+      },
+      updatedTodo: {
+        description: '',
+      },
+      visibility: false,
+      clicked: ''
+    }
+
+    this.changeTodoStage = this.changeTodoStage.bind(this)
+    this.captureUserInput = this.captureUserInput.bind(this)
+    this.onFormSubmit = this.onFormSubmit.bind(this)
+    this.deleteTodo = this.deleteTodo.bind(this)
+    this.updateTodo = this.updateTodo.bind(this)
+    this.changeButtonsVisibility = this.changeButtonsVisibility.bind(this)
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({ todos })
+    }, 1000)
+  }
+
+  changeTodoStage(todoStage, todoID) {
```

Comment:
Nice that you used this function! This is quite similar with `updateTodo`. The only thing that changes is `{ stage: todoStage }`. Maybe you can generalize this one and end up with one function?

Reply:
now that I see it again I see what you mean